File: Ser2023/Company/Company.Web/Modules/Store/ProductLang/ProductLangRow.cs
using Serenity.ComponentModel;
using Serenity.Data;
using Serenity.Data.Mapping;
using System.ComponentModel;

namespace Company.Store;

[ConnectionKey("Store"), Module("Store"), TableName("[dbo].[ProductLang]")]
[DisplayName("Product Lang"), InstanceName("Product Lang")]
[ReadPermission("Store:General")]
[ModifyPermission("Store:General")]
public sealed class ProductLangRow : Row<ProductLangRow.RowFields>, IIdRow, INameRow, ILocalizationRow
{
    [DisplayName("Id"), Column("ID"), Identity, IdProperty]
    public int? Id { get => fields.Id[this]; set => fields.Id[this] = value; }

    [DisplayName("Product Id"), Column("ProductID"), NotNull]
    public int? ProductId { get => fields.ProductId[this]; set => fields.ProductId[this] = value; }

    [DisplayName("Language Id"), Column("LanguageID"), NotNull]
    public int? LanguageId { get => fields.LanguageId[this]; set => fields.LanguageId[this] = value; }

    [DisplayName("Product Name"), Size(40), QuickSearch, NameProperty]
    public string ProductName { get => fields.ProductName[this]; set => fields.ProductName[this] = value; }
    public Field CultureIdField { get => Fields.LanguageId; }

    public class RowFields : RowFieldsBase
    {
        public Int32Field Id;
        public Int32Field ProductId;
        public Int32Field LanguageId;
        public StringField ProductName;
    }
}

File: Ser2023/Company/Company.Web/Modules/ServerTypes/Store/ProductLangRow.ts
import { fieldsProxy } from "@serenity-is/corelib/q";

export interface ProductLangRow {
    Id?: number;
    ProductId?: number;
    LanguageId?: number;
    ProductName?: string;
}

export abstract class ProductLangRow {
    static readonly idProperty = 'Id';
    static readonly nameProperty = 'ProductName';
    static readonly localTextPrefix = 'Store.ProductLang';
    static readonly deletePermission = 'Store:General';
    static readonly insertPermission = 'Store:General';
    static readonly readPermission = 'Store:General';
    static readonly updatePermission = 'Store:General';

    static readonly Fields = fieldsProxy<ProductLangRow>();
}